clients/drcachesim/tracer/kimage.h

```diff
@@ -0,0 +1,68 @@
+/* **********************************************************
+ * Copyright (c) 2022 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* kimage.h: kimage is a file that contains all kernel code segments. It is used to decode
+ * kernel PT trace. The format of the file is:
+ *  ------------------------------------------------
+ * | An instance of kimage_hdr_t                    |
+ *  ------------------------------------------------
+ * | A table of kimage_code_segment_hdr_t instances |
+ *  ------------------------------------------------
+ * | Binary data for all code segments              |
+ *  ------------------------------------------------
```

Comment:
I think using ELF format wouldn't be much more complex in the code (it would simplify it in some ways since this header would disappear) and might be easier to debug since we could use standard ELF tools to examine the binary file.  You already have this format which seems ok but if you hadn't yet written it I would have voted for just using the standard ELF format.

Reply:
If we output in ELF format would the need for the -image_format and separate handling for this custom format all disappear and everything would be handled by the existing ELF file parsing?  If so, I would vote for that and not using a custom format as I don't see it adding any significant advantage and in fact it is adding complexity by adding this separate type to handle.